packages/api/endpoints/workflows.js

```diff
@@ -26,20 +30,28 @@ function list(event, cb) {
  * @param {callback} cb - aws lambda callback function
  * @returns {undefined} undefined
  */
-function get(event, cb) {
+async function get(event, cb) {
   const name = _get(event.pathParameters, 'name');
 
-  const key = `${process.env.stackName}/workflows/list.json`;
-  S3.get(process.env.bucket, key)
-    .then((file) => {
-      const workflows = JSON.parse(file.Body.toString());
+  const workflowKey = `${process.env.stackName}/workflows/list.json`;
+  try {
+    const { Body } = await aws.getS3Object(process.env.bucket, workflowKey);
+
+    const jsonResponse = JSON.parse(Body);
 
-      const matchingWorkflow = workflows.find((workflow) => workflow.name === name);
-      if (matchingWorkflow) return cb(null, matchingWorkflow);
+    const matchingWorkflow = jsonResponse.find((workflow) => workflow.name === name);
+    if (matchingWorkflow) return cb(null, matchingWorkflow);
 
-      return cb({ message: `A record already exists for ${name}` });
-    })
-    .catch(cb);
+    const e = new Error('The specified key does not exist.');
+    e.name = 'NoSuchKey';
+    throw e;
```

Comment:
Instead of creating a `NoSuchKey` error here, this should instead say

```javascript
return cb(err, null, 404);
```

It's confusing to see a NoSuchKey error when there _is_ such a key.

Reply:
Changed, but I did make an error but err was undefined at that point